File: solutions/java/collatz-conjecture/1/src/main/java/CollatzCalculator.java
class CollatzCalculator {

  int computeStepCount(int start) {
    if (start > 0) return collatz(start);
    throw new IllegalArgumentException("Only positive integers are allowed");
  }

  int collatz(int n) {
    return n > 1 ? 1 + collatz(n % 2 == 0 ? n / 2 : 3 * n + 1) : 0;
  }
}

File: solutions/kotlin/collatz-conjecture/1/src/main/kotlin/CollatzCalculator.kt
object CollatzCalculator {
    fun computeStepCount(start: Int): Int {
        require(start > 0)
        tailrec fun go(n: Int, acc: Int): Int =
            if (n == 1) acc
            else if (n % 2 == 0) go(n / 2, acc + 1)
            else go(3 * n + 1, acc + 1)
        return go(start, 0)
    }
}
